lib/chef/resource/package.rb

```diff
@@ -31,14 +31,13 @@ class Package < Chef::Resource
 
       def initialize(name, run_context=nil)
         super
-        @candidate_version = nil
-        @options = nil
-        @package_name = name
-        @response_file = nil
-        @response_file_variables = Hash.new
-        @source = nil
-        @version = nil
-        @timeout = nil
+        options                 nil
+        package_name            name
+        response_file           nil
+        response_file_variables Hash.new
+        source                  nil
+        version                 nil
+        timeout                 nil
```

Comment:
@danielsdeleo how about this?  the annoying thing is that when people bypass the accessor for non-nil values in the initializer it can cause problems (I specifically need to set the package_name this way in order to be able to subclass and then coerce it into an array--code not shown in this commint).  there was also the case of the log resource when people started calling the log resource with random objects and the initializer bypassed the strict type checking of set_or_return and so stored random non-string classes internally in the ivar.

the problem with what i'm doing though is that i'm still not explicitly setting nil because we can't have nillable properties until Chef 13, and relying on the getter to create the ivar.  that might still throw a warning somewhere under set_or_return, but we could fix it there.